### PR TITLE
[SPARK-45515][CORE][SQL] Use enhanced `switch` expressions to replace the regular `switch` statement

### PR DESCRIPTION
**Pull request by LuciferYang**
### What changes were proposed in this pull request?
This pr use enhanced  `switch` Expressions to replace the regular `switch` statement in Spark Java code refer to [JEP 361](https://openjdk.org/jeps/361)

Example:

```java
double getPrice(String fruit) {
  switch (fruit) {
    case "Apple":
      return 1.0;
    case "Orange":
      return 1.5;
    case "Mango":
      return 2.0;
    default:
      throw new IllegalArgumentException();
   }
 }
```  

Can be changed to 

```java
double getPrice(String fruit) {
  return switch (fruit) {
    case "Apple" -> 1.0;
    case "Orange" -> 1.5;
    case "Mango" -> 2.0;
    default -> throw new IllegalArgumentException();
  };
}
```

This pr does not include parts of the `hive-thriftserver` module.

### Why are the changes needed?
Using `JEP 361: Switch Expressions` can bring the following benefits:

1. **More concise syntax**: `switch` can be used as an expression, not just a statement. This makes the code more concise and easier to read.

2. **Safer**: In `switch` expressions, if we forget the `break`, there will be no unexpected `fall-through` behavior. At the same time, the compiler will check whether all possible cases are covered. If not all cases are covered, the compiler will report an error.

3. **Easier to understand**: The new `switch` expression syntax is closer to our decision-making pattern in daily life, making the code easier to understand.

### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?
Pass GitHub Actions

### Was this patch authored or co-authored using generative AI tooling?
No

### CODE REVIEW

**Comment by LuciferYang** [comment]
[JEP 361](https://openjdk.org/jeps/361)

**Review by srowen — APPROVED** [review]

**Comment by LuciferYang** [comment]
Merged into master, thanks @srowen

File: common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java [hunk 1]
```diff
@@ -65,38 +65,26 @@ public static ThreadFactory createThreadFactory(String threadPoolPrefix) {
...
-      case NIO:
-        return new NioEventLoopGroup(numThreads, threadFactory);
-      case EPOLL:
-        return new EpollEventLoopGroup(numThreads, threadFactory);
-      default:
-        throw new IllegalArgumentException("Unknown io mode: " + mode);
-    }
+    return switch (mode) {
+      case NIO -> new NioEventLoopGroup(numThreads, threadFactory);
+      case EPOLL -> new EpollEventLoopGroup(numThreads, threadFactory);
```
Comment (beliefer) [inline common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java:70]: It seems not treat the default path.

Comment (beliefer) [inline common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java:70]: It seems not treat the default path.

Comment (LuciferYang) [reply common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java:70]: image

Comment (LuciferYang) [reply common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java:70]: Additionally, if IOMode is extended and this method is not updated in sync, a compile error will occur to remind the developer to add the corresponding case branch, so, for this enum matching scenario, `default` is not necessary:)

Comment (beliefer) [reply common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java:70]: Yeah. Thank you for the explanation. The origin behavior will throw exception. What's now?

Comment (LuciferYang) [reply common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java:70]: Compilation failed if we just add a new IOMode item, like 

```
[error] /Users/yangjie01/SourceCode/git/spark-mine-sbt/common/network-common/src/main/java/org/apache/spark/network/util/NettyUtils.java:68:1:  错误: switch 表达式不包含所有可能的输入值
[error]     return switch (mode) {
[error]            ^
[warn] javac exited with exit code 1
[error] (network-common / Compile / compileIncremental) javac returned non-zero exit code
[error] Total time: 7 s, completed 2023年10月13日 下午3:42:54

```

**Pull request closed without merging**